packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Models/Types/ModelTypeProvider.cs

```diff
@@ -121,5 +121,203 @@ private bool PropertyHasSetter(CSharpType type, InputModelProperty prop)
 
             return null;
         }
+
+        protected override CSharpMethod[] BuildConstructors()
+        {
+            List<CSharpMethod> constructors = new List<CSharpMethod>();
+
+            var initializationConstructor = BuildInitializationConstructor();
+            var skipInitializerConstructor = _inputModel.IsUnknownDiscriminatorModel;
+            if (!skipInitializerConstructor)
+            {
+                constructors.Add(initializationConstructor);
+            }
+
+            var serializationConstructor = BuildSerializationConstructor();
+            bool serializationParametersMatchInitialization = serializationConstructor != null &&
+                !serializationConstructor.Signature.Parameters.Any(p => p.Type.IsList) &&
+                initializationConstructor.Signature.Parameters.SequenceEqual(serializationConstructor.Signature.Parameters, Parameter.EqualityComparerByType);
+
+            if (serializationConstructor != null && !serializationParametersMatchInitialization)
+            {
+                constructors.Add(serializationConstructor);
+            }
+
+            var initCtorParameterCount = skipInitializerConstructor ? int.MaxValue : initializationConstructor.Signature.Parameters.Count; // if the ctor is skipped, we return a large number to avoid the case that the skipped ctor has 0 parameter.
```

Comment:
well this reads hacky.
Could we make the initializationConstructor to be nullable depending on the value of `skipInitializerConstructor` (similar to the serializationConstructor part), and we could have
```csharp
if (initializationConstructor?.Signature.Parameters.Count > 0) // this gives us false if the ctor is null
```